elastic/kibana
# lens esql generation

## Summary

updates lens so it uses ESQL when requesting data from elasticsearch rather than DSL when that is possible.

nothing should change for the user, everything should work exactly as before

but when going to inspector you will be able to see that in some cases we will send out ESQL request.

the feature is only enabled when the `lens.enable_esql` feature flag is enabled.

at the moment this covers very few cases, but rather builds the necessary infrastructure so we can keep improving this over time:
- user has to be in UTC timezone
- user needs to use set of lens operations that are supported



### how to test:

1. put this in your `kibana.dev.yaml`:
```
feature_flags.overrides:
  lens.enable_esql: true
```
2. go to advanced settings and set timezone to UTC
3. go to lens, configure a chart (for example a metric chart showing average of `products.price`
4. go to inspector -> requests and you can see that we sent out ESQL query
5. visually nothing should change

### todo:

- [x] make sure CI is green when the feature flag is enabled
